grin/src/Transformations/ExtendedSyntax/Conversion.hs

```diff
@@ -125,16 +125,40 @@ instance Convertible Exp New.Exp where
   convert exp = fst $ evalNameM exp $ flip anaM exp $ \case
     (Program exts defs)  -> pure $ New.ProgramF (map convert exts) defs
     (Def name args body) -> pure $ New.DefF (convert name) (map convert args) body
-    {- NOTE: we assume Binding Pattern Simplification has been run
+
+    {- NOTE: We assume Binding Pattern Simplification has been run,
```

Comment:
Is there an easy way to check that assumption?

Reply:
Looking at the code snippet it is not clear, why it does the trick. Please could you explain it?

Reply:
This function is only called after `nameEverything`. The other cases are deliberately left out to get a runtime error, so we know that the conversion failed.

Reply:
I added a more detailed error message to the transformation.